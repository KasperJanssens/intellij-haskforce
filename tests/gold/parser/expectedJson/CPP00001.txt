[{"Module":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":7,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":1,"srcSpanStartLine":7,"srcSpanEndLine":7,"srcSpanStartColumn":1}]},{"ModuleHead":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":7,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":22,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":17}]},{"ModuleName":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":16,"srcSpanStartLine":2,"srcSpanEndLine":2,"srcSpanStartColumn":8},"srcInfoPoints":[]},"CPP00001"]},null,null]},[{"LanguagePragma":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},"srcInfoPoints":[{"srcSpanFilename":"A.hs","srcSpanEndColumn":13,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":1},{"srcSpanFilename":"A.hs","srcSpanEndColumn":21,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":18}]},[{"Ident":[{"srcInfoSpan":{"srcSpanFilename":"A.hs","srcSpanEndColumn":17,"srcSpanStartLine":1,"srcSpanEndLine":1,"srcSpanStartColumn":14},"srcInfoPoints":[]},"CPP"]}]]}],[],[]]},[]]