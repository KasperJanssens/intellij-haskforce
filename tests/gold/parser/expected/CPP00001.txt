Haskell File
  HaskellPpragmaImpl(PPRAGMA)
    PsiElement(HaskellTokenType.{-#)('{-#')
    PsiElement(HaskellTokenType.PRAGMA)(' LANGUAGE CPP ')
    PsiElement(HaskellTokenType.#-})('#-}')
  HaskellModuledeclImpl(MODULEDECL)
    PsiElement(HaskellTokenType.module)('module')
    HaskellQconidImpl(QCONID)
      HaskellConidImpl(CONID)
        PsiElement(HaskellTokenType.conidRegexp)('CPP00001')
    PsiElement(HaskellTokenType.where)('where')
  PsiComment(HaskellTokenType.CPPIF)('#if __GLASGOW_HASKELL__ < 708')
  PsiComment(HaskellTokenType.CPPDEFINE)('#define DIFFERENT_MONADIO 1')
  PsiComment(HaskellTokenType.CPPDEFINE)('#define MONADIO_INSTANCES 1')
  PsiComment(HaskellTokenType.#endif)('#endif')
  HaskellBodyImpl(BODY)
    <empty list>